### PR TITLE
Feature/17383 speech synthesis on questions

### PR DESCRIPTION
**Pull request by jon-shipley**
Add speech synthesis to the question.  The timer starts counting down straight after the question has been asked.

### CODE REVIEW

File: pupil-spa/src/app/check/check.component.ts [hunk 1]
```diff
@@ -149,7 +153,16 @@ export class CheckComponent implements OnInit {
...
         this.question = this.warmupQuestionService.getQuestion(parseInt(matches[ 1 ], 10));
-        this.viewState = 'question';
+        this.viewState = 'practice-question';
+        break;
+      }
+      case CheckComponent.spokenWarmupQuestionRe.test(stateDesc): {
+        // Show the warmup question screen
+        const matches = CheckComponent.spokenWarmupQuestionRe.exec(stateDesc);
+        this.isWarmUp = true;
+        // console.log(`state: ${stateDesc}: question is ${matches[ 1 ]}`);
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/check/check.component.ts]: do you still need that comment?

Comment (jon-shipley) [reply pupil-spa/src/app/check/check.component.ts]: Yes, will convert these console.log to logger entries at some point soon.

File: pupil-spa/src/app/practice-question/practice-question.component.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+      }
+      this.remainingTime = Math.ceil(timeLeft);
+    }, 100);
+  }
+
+  /**
+   * Check a manual submission to see if it is allowed.
+   * @return {boolean}
+   */
+  answerIsLongEnoughToManuallySubmit() {
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/practice-question/practice-question.component.ts]: A bit confusing naming convention in my opinion.
Shouldn't that be **hasAnswer**?

File: pupil-spa/src/app/services/speech/speech.service.ts [hunk 3]
```diff
@@ -25,13 +48,32 @@ export class SpeechService {
    * @param utterance
    */
   speak(utterance: string): void {
+    this.cancel();
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/services/speech/speech.service.ts]: is cancelled needed at the beginning? what happens if you remove it?

Comment (jon-shipley) [reply pupil-spa/src/app/services/speech/speech.service.ts]: cancel() ensures that any speech in progress will be curtailed immediately.  It's a preventative measure.

**Pull request merged**